docs/IR.md

```diff
@@ -1,127 +1,372 @@
+Attributes, which represent literal constants passed to operators from nodes, have the following properties:
+
+Name|Type|Description
+|---|---|---|
+name|string|The name of the attribute. Must be unique among attributes, inputs, and outputs for any given operator and node.
+doc_string|string|A human-readable documentation for this value. Markdown is allowed.
+type|AttributeType|The type of the attribute, determining which of the remaining fields is used to hold the value of the attribute.
+f|float|A floating-point value.
+i|int64|An integer value
+S|byte[]|UTF-8 string
+t|Tensor|A tensor value
+g|Graph|A graph
+floats|float[]|List of floats
+ints|int64[]|List of integers
+strings|byte[][]|List of strings
+tensors|Tensor[]|List of tensor values
+graphs|Graph[]|List of graphs
+
+The properties ‘name’ and ‘type’ are required on all attributes, and ‘doc_string’ SHOULD be used on all attributes. An attribute MUST have only one of the value-carrying properties.
 
 #### Values
 
-The representation distinguishes between two kinds of values: attribute values, which are statically known, and runtime values. The type of values permitted in the two cases are different. The permitted types of attribute values are indicated by the enumeration `AttributeType`, while the permitted types of runtime values are described by `TypeProto`. 
+The representation distinguishes between two kinds of values: attribute values, which are statically known, and runtime values. The types of values permitted in the two cases are different.
 
-The types of the inputs and outputs of the model must be specified, including the shapes of tensors. While the ranks of input and output tensors are statically specified, the sizes of specific dimensions (axis) may be statically unknown and are indicated so using symbolic identifiers in the shape. 
+The types of the inputs and outputs of a graph must be specified.
 
-#### Optional Inputs
+#### Variadic Inputs and Outputs
+ 
+The last input or output of an operator MAY be marked as variadic. For example, the operator 'Max()' can be used to compute the maximum of a varying number of input values.
 
-Some Operators have inputs that are marked as optional. There are two ways to leave an optional input unspecified. The first is to simply not provide that input. However, this is not always possible - for example, if you wish to leave the fourth input unspecified, but still provide a value for the fifth input. Therefore, any input with a name of the empty string is treated as an unspecified optional input.
+For each variadic operator input, one or more node inputs must be specified. For each variadic operator output, one or more node outputs must be specified. 
 
-Built-in Operators and Standard Data Types
-------------------------------------------
+#### Optional Inputs and Outputs
 
-### Operators
+Some operators have inputs that are marked as optional, which means that a referring node MAY choose not to provide a value for that input.
```

Comment:
"a referring node MAY choose not to provide a value for that input" or "a user of such an operator MAY choose not to provide values for some of optional inputs"?